lib/plugins/aws/lib/naming.js

```diff
@@ -0,0 +1,234 @@
+          'lambda',
+        ],
+      ],
+    };
+  }
+  getLogicalPolicyName() {
+    return 'IamPolicyLambdaExecution';
+  }
+
+  getLogGroupName(functionName) {
+    return `/aws/lambda/${functionName}`;
+  }
+
+  /*
+   * Lambda Function Naming
+   */
+  getNormalizedLambdaName(functionName) {
+    return this.normalizeName(functionName);
+  }
+  extractLambdaNameFromArn(functionArn) {
+    const splitArn = functionArn.split(':');
+    // TODO the following two lines assumes default function naming?  Is there a better way?
+    // TODO (see ~/lib/classes/Service.js:~155)
+    const splitName = splitArn[splitArn.length - 1].split('-');
+    return splitName[splitName.length - 1];
+  }
+  // TODO rectify this with the above and the code that uses both
+  // TODO (see ~/lib/plugins/aws/info/index.js)
+  extractLambdaNameFromArn2(functionArn) {
+    return functionArn.substring(functionArn.lastIndexOf(':') + 1);
+  }
+  getLogicalLambdaName(functionName) {
+    return `${this.getNormalizedLambdaName(functionName)}LambdaFunction`;
+  }
+  getLogicalLambdaNameRegex() {
+    return /LambdaFunction$/;
+  }
+  getLogicalLambdaArnName(functionName) {
+    return `${this.getLogicalLambdaName(functionName)}Arn`;
+  }
+  getLogicalLambdaArnNameRegex() {
+    return /LambdaFunctionArn$/;
+  }
+
+  /*
+   * ApiGateway Authorizer Lambda & Method Naming
+   */
+  getApiGatewayName() {
+    return `${this.serverless.config.stage}-${this.serverless.service.service}`;
+  }
+  getLogicalApiGatewayName() {
+    return 'ApiGatewayRestApi';
+  }
+  getNormalizedAuthorizerName(functionName) {
+    return this.getNormalizedLambdaName(functionName);
+  }
+  getLogicalAuthorizerName(functionName) {
+    return `${this.getNormalizedAuthorizerName(functionName)}ApiGatewayAuthorizer`;
+  }
+  extractAuthorizerIdFromArn(authorizerArn) {
+    return this.extractLambdaNameFromArn(authorizerArn);
+  }
+  getLogicalAuthorizerArnName(functionName) {
+    return this.getLogicalLambdaArnName(functionName);
+  }
+  getNormalizedApiGatewayResourceName(resourcePath) {
+    return resourcePath.split('/').map(
+      this.normalizePathtoCapitalAlphaNumbericOnlyWithReplacement.bind(this)
+    ).join('');
+  }
+  getLogicalApiGatewayResourceName(resourcePath) {
+    return `ApiGatewayResource${this.getNormalizedApiGatewayResourceName(resourcePath)}`;
+  }
+  extractResourceId(logicalApiGatewayResourceName) {
+    return logicalApiGatewayResourceName.match(/ApiGatewayResource(.*)/)[1];
+  }
+  getNormalizedApiGatewayMethodName(methodName) {
+    return this.normalizeName(methodName.toLowerCase());
+  }
+  getLogicalApiGatewayMethodName(resourceId, methodName) {
+    return `ApiGatewayMethod${resourceId}${this.getNormalizedApiGatewayMethodName(methodName)}`;
+  }
+  getLogicalApiGatewayApiKeyRegex() {
+    return /^ApiGatewayApiKey/;
+  }
+
+  /*
+   * S3 Bucket Naming
+   */
+  getLogicalDeploymentBucketName() {
+    return 'ServerlessDeploymentBucket';
```

Comment:
would that work with the recent custom bucket feature we merged few days ago? we might need another method or something and this one could be called `getLogicalDeploymentBucketName`